src/commands/flutter-symbols/upload.ts

```diff
@@ -0,0 +1,468 @@
+      uploadInfo.push({
+        fileType: 'dSYMs',
+        location: this.iosDsymsLocation!,
+        platform: 'ios',
+      })
+    }
+    if (this.androidMappingLocation) {
+      uploadInfo.push({
+        fileType: 'Proguard Mapping File',
+        location: this.androidMappingLocation!,
+        platform: 'Android',
+      })
+    }
+    if (this.dartSymbolsLocation) {
+      uploadInfo.push({
+        fileType: 'Dart Symbol Files',
+        location: this.dartSymbolsLocation!,
+        platform: 'Flutter',
+      })
+    }
+
+    this.context.stdout.write(renderCommandInfo(this.dryRun, this.version!, this.serviceName, this.flavor, uploadInfo))
+
+    this.config = await resolveConfigFromFile(this.config, {
+      configPath: this.configPath,
+      defaultConfigPath: DEFAULT_CONFIG_PATH,
+    })
+
+    if (!this.disableGit) {
+      this.gitData = await this.getGitMetadata()
+    }
+
+    const metricsLogger = getMetricsLogger({
+      apiKey: this.config.apiKey,
+      datadogSite: this.config.datadogSite,
+      defaultTags: [
+        `version:${this.version}`,
+        `service:${this.serviceName}`,
+        `cli_version:${this.cliVersion}`,
+        'flutter:true',
+        'platform:android',
+      ],
+      prefix: 'datadog.ci.symbols.upload.',
+    })
+
+    const apiKeyValidator = newApiKeyValidator({
+      apiKey: this.config.apiKey,
+      datadogSite: this.config.datadogSite,
+      metricsLogger: metricsLogger.logger,
+    })
+
+    const initialTime = Date.now()
+
+    const callResults: UploadStatus[] = []
+    try {
+      if (this.iosDsymsLocation) {
+        callResults.push(await this.performDsymUpload())
+      }
+      if (this.androidMappingLocation) {
+        callResults.push(await this.performAndroidMappingUpload(metricsLogger, apiKeyValidator))
+      }
+      if (this.dartSymbolsLocation) {
+        callResults.push(...(await this.performDartSymbolsUpload(metricsLogger, apiKeyValidator)))
+      }
+
+      const totalTime = (Date.now() - initialTime) / 1000
+      this.context.stdout.write(renderCommandSummary(callResults, totalTime, this.dryRun))
+    } catch (e) {
+      this.context.stderr.write(renderGeneralizedError(e))
+
+      return 1
+    } finally {
+      try {
+        await metricsLogger.flush()
+      } catch (err) {
+        this.context.stdout.write(`WARN: ${err}\n`)
+      }
+    }
+
+    return 0
+  }
+
+  private getAndroidMetadata(): MappingMetadata {
+    return this.getMappingMetadata(TYPE_JVM_MAPPING)
+  }
+
+  private getFlutterMetadata(platform: string, arch: string) {
+    return this.getMappingMetadata(TYPE_DART_SYMBOLS, platform, arch)
+  }
+
+  private getFlutterSymbolFiles(dartSymbolLocation: string): string[] {
+    const symbolPaths = glob.sync(buildPath(dartSymbolLocation, '*.symbols'))
+
+    return symbolPaths
+  }
+
+  private getGitDataPayload(): MultipartValue | undefined {
+    if (this.gitData === undefined) {
+      return undefined
+    }
```

Comment:
This check is already performed before each call of `getGitDataPayload`.
You could pass `this.gitData` as an argument, avoiding to check for its definition here and prevent the function from possibly returning `undefined` (which in turn allows to remove the `!` when [setting the payload](https://github.com/DataDog/datadog-ci/pull/616/files#diff-3459254cea55c76190454bf3617bfea34521699313c555379c3519d3c3328732R303)).

More generally: we try to avoid using `!` as much as possible as it means we are forcing the type checking of typescript to believe something that "should" be obvious for typescript when parsing the code.